packages/kit/src/runtime/client/client.js

```diff
@@ -1376,6 +1376,19 @@ export function create_client(app, target) {
 		},
 
 		goto: (href, opts = {}) => {
+			if (typeof href === 'string') {
+				href = new URL(href, get_base_uri(document));
+			}
+			if (href.origin !== origin) {
+				return Promise.reject(
+					new Error(
+						DEV
+							? `Cannot use \`goto\` with an external URL. Use \`window.location = "${href}"\` instead`
+							: 'Invalid URL'
```

Comment:
does this need to include that `goto` was called?
```suggestion
							: 'goto: invalid URL'
```